riscv-c-api.md

```diff
@@ -114,7 +129,92 @@ For example:
+
+The extensions `Zkt` or `Zkr` do not currently introduce intrinsics.
+Data-independent latency is a property of the implementation, while
+`Zkr` (the entropy source interface) is implemented as a CSR only
+(`seed` at CSR address `0x015`).
+
+Many of the `Zbkb` instructions can be inferred compilers so they do not
+have direct intrinsics equivalents. Rotations are included mainly as RISC-V
+sign extension behavior, and clamping of rotation amount may generate extra
+instructions unless the programmer is very careful.
+
+### Scalar Cryptography Extension short form intrinsics (alphabetically)
+
+| Prototype                                                     | Mnemonic      | Short Description                         | Supported in                  |
+| ------------------------------------------------------------- | ------------- | ----------------------------------------- | ----------------------------- |
+| `int32_t _rv32_aes32dsi(int32_t rs1, int32_t rs2, int bs);`   | `aes32dsi`    | AES final round decryption / RV32.        | Zknd, Zkn, Zk (RV32)          |
+| `int32_t _rv32_aes32dsmi(int32_t rs1, int32_t rs2, int bs);`  | `aes32dsmi`   | AES middle round decryption / RV32.       | Zknd, Zkn, Zk (RV32)          |
+| `int32_t _rv32_aes32esi(int32_t rs1, int32_t rs2, int bs);`   | `aes32esi`    | AES final round encryption / RV32.        | Zkne, Zkn, Zk (RV32)          |
+| `int32_t _rv32_aes32esmi(int32_t rs1, int32_t rs2, int bs);`  | `aes32esmi`   | AES middle round encryption / RV32.       | Zkne, Zkn, Zk (RV32)          |
+| `int64_t _rv64_aes64ds(int64_t rs1, int64_t rs2);`            | `aes64ds`     | AES final round decryption / RV64.        | Zknd, Zkn, Zk (RV64)          |
+| `int64_t _rv64_aes64dsm(int64_t rs1, int64_t rs2);`           | `aes64dsm`    | AES middle round decryption / RV64        | Zknd, Zkn, Zk (RV64)          |
+| `int64_t _rv64_aes64es(int64_t rs1, int64_t rs2);`            | `aes64es`     | AES final round encryption / RV64.        | Zkne, Zkn, Zk (RV64)          |
+| `int64_t _rv64_aes64esm(int64_t rs1, int64_t rs2);`           | `aes64esm`    | AES middle round encryption / RV64.       | Zkne, Zkn, Zk (RV64)          |
+| `int64_t _rv64_aes64im(int64_t rs1);`                         | `aes64im`     | AES Inverse MixColumns, key schedule.     | Zknd, Zkn, Zk (RV64)          |
+| `int64_t _rv64_aes64ks1i(int64_t rs1, int rnum);`             | `aes64ks1i`   | AES key schedule, round number.           | Zkne, Zknd, Zkn, Zk (RV64)    |
+| `int64_t _rv64_aes64ks2(int64_t rs1, int64_t rs2);`           | `aes64ks2`    | AES key schedule, word mixing.            | Zkne, Zknd, Zkn, Zk (RV64)    |
+| `int32_t _rv32_brev8(int32_t rs1);`                           | `brev8`       | Reverse order of bits within each byte.   | Zbkb (RV32)                   |
+| `int64_t _rv64_brev8(int64_t rs1);`                           | `brev8`       | Reverse order of bits within each byte.   | Zbkb (RV64)                   |
```

Comment:
Maybe provide `long _rv_brev8(long rs1)` only?

Reply:
It's a RISC-V convention to sign extend all 32-bit "W" intrinsics; even things like 32-bit rotation on RV64. The architecture review committee asked the SHA256 instructions to be like that too, for consistency, so they are like that. Everything is sign-exteded -- Indeed this trips a lot programmers. Also, what's precisely why the (signed) "long" type was used in bitmanip (they were all sign extended at least since 2019 -- so not as you recall). If one uses unsigned types then there is a risk of automatic generation of instructions that clear the upper half.

Reply:
I am not sure I've got the point about sign extension? long type in kind of more `intXLEN_t`, so there is no sign-extension issue on RV64 I think?

Reply:
I guess I only thinking about the interface itself is equivalence and not consider the semantic/function is different on RV32/RV64, different name/interface type should be more clear to let user aware that, I am OK with current design.